pkg/apis/config/defaults.go

```diff
@@ -70,34 +70,69 @@ func NewDefaultsConfigFromConfigMap(config *corev1.ConfigMap) (*Defaults, error)
 type Defaults struct {
 	// NamespaceDefaultsConfig are the default Broker Configs for each namespace.
 	// Namespace is the key, the value is the KReference to the config.
-	NamespaceDefaultsConfig map[string]*ClassAndKRef `json:"namespaceDefaults,omitempty"`
+	NamespaceDefaultsConfig map[string]*ClassAndBrokerConfig `json:"namespaceDefaults,omitempty"`
 
 	// ClusterDefaultBrokerConfig is the default broker config for all the namespaces that
 	// are not in NamespaceDefaultBrokerConfigs.
-	ClusterDefault *ClassAndKRef `json:"clusterDefault,omitempty"`
+	ClusterDefault *ClassAndBrokerConfig `json:"clusterDefault,omitempty"`
 }
 
-// ClassAndKRef contains configuration for a given namespace for broker. Allows
-// configuring both the Class of the Broker as well as the reference to the
-// config it should use
-type ClassAndKRef struct {
-	BrokerClass        string `json:"brokerClass,omitempty"`
+// ClassAndBrokerConfig contains configuration for a given namespace for broker. Allows
+// configuring the Class of the Broker, the reference to the
+// config it should use and it's delivery.
+type ClassAndBrokerConfig struct {
+	BrokerClass   string `json:"brokerClass,omitempty"`
+	*BrokerConfig `json:",inline"`
+}
+
+// BrokerConfig contains configuration for a given namespace for broker. Allows
+// configuring the reference to the
+// config it should use and it's delivery.
+type BrokerConfig struct {
 	*duckv1.KReference `json:",inline"`
+	Delivery           *DeliveryConfig `json:"delivery,omitempty"`
```

Comment:
Instead of re-defining the DeliveryConfig here, could we just use the definition from pkg/apis/duck/v1/delivery_types?

Reply:
If I use v1Delivery, I need to convert v1Delivery to v1beta1Delivery.

Reply:
I'm sorry, I don't quite understand. Do you mean you need to add conversion methods, or?

Reply:
yes i need add conversion methods
https://github.com/knative/eventing/blob/master/pkg/apis/duck/v1/delivery_conversion.go#L28

Reply:
Ah, yeah, it just means that you can not convert anything higher. You can always convert v1->v1beta1 or just use v1. Just like all the other conversions (say, eventing/v1/broker_conversion), and in eventing/v1beta1/broker_conversion we know how to go from v1->v1beta1 and v1beta1->v1.
Does that help?

Reply:
thanks, i will use pkg/apis/duck/v1/delivery_types